examples/platform/silabs/efr32/rs911x/hal/efx_spi.c

```diff
@@ -195,21 +337,22 @@ int16_t rsi_spi_transfer(uint8_t * tx_buf, uint8_t * rx_buf, uint16_t xlen, uint
         return RSI_ERROR_SPI_BUSY;
     }
 
-    configASSERT(spiInitiatorTaskHandle == NULL); // No other task should currently be waiting for the dma completion
+    // No other task should currently be waiting for the dma completion
+    configASSERT(spiInitiatorTaskHandle == NULL);
```

Comment:
Weird and also false. the UART task can be waiting for the DMA completion interrupt.

Reply:
Ah that's my code so I can explain. The comment is indeed not that clear. 

This is specific to the `SPI_DRV` and the `rsi_spi_transfer`. Basically, we capture the task handle of the task calling for this spi transfer (next line). Only 1 task at a time can own the DMA transfer as the DMA will notify that specific task that it has been completed. There is semaphore protection in place to prevent that from happening. But I did still add this assert just it case, something weird happen and I missed something. It is just additional protection and it doesn't have to do with other peripherals using the DMA